server/src/main/java/org/apache/druid/rpc/guice/ServiceClientModule.java

```diff
@@ -117,4 +115,11 @@ public CoordinatorClient makeCoordinatorClient(
         jsonMapper
     );
   }
+
+  public static ServiceClientFactory getServiceClientFactory(@EscalatedGlobal final HttpClient httpClient)
```

Comment:
Super nit: Probably should be named the other way around.
This method should be called `makeServiceClientFactory` and the other one `getServiceClientFactory`.

Reply:
Ha, done. I also added Guice module tests to verify these instances are provided, which should also address CI failures due to insufficient coverage.